src/main/java/hudson/plugins/release/pipeline/ReleaseStep.java

```diff
@@ -0,0 +1,218 @@
+                } else {
+                    throw new AbortException("Job doesn't have release plugin configuration");
+                }
+            }
+            return parameters;
+        }
+
+        @Override
+        public boolean start() throws Exception {
+            if (step.getJob() == null) {
+                throw new AbortException("Job name is not defined.");
+            }
+
+            final AbstractProject project = Jenkins.getActiveInstance()
+                                                   .getItem(step.getJob(), invokingRun.getParent(), AbstractProject.class);
+            if (project == null) {
+                throw new AbortException("No parametrized job named " + step.getJob() + " found");
+            }
+            listener.getLogger().println("Releasing project: " + ModelHyperlinkNote.encodeTo(project));
+
+            List<Action> actions = new ArrayList<>();
+            StepContext context = getContext();
+            actions.add(new ReleaseTriggerAction(context));
+            LOGGER.log(Level.FINER, "scheduling a release of {0} from {1}", new Object[] { project, context });
+            actions.add(new SafeParametersAction(updateParametersWithDefaults(project, step.getParameters())));
+            actions.add(new ReleaseWrapper.ReleaseBuildBadgeAction());
+
+            QueueTaskFuture<?> task = project.scheduleBuild2(0, new Cause.UpstreamCause(invokingRun), actions);
+            if (task == null) {
+                throw new AbortException("Failed to trigger build of " + project.getFullName());
+            }
+
+            return false;
+        }
+
+        @Override
+        public void stop(@Nonnull Throwable cause) throws Exception {
+            getContext().onFailure(cause);
+        }
+
+        private static final long serialVersionUID = 1L;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "release";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Trigger release for the job";
+        }
+
+        @Override public Step newInstance(StaplerRequest req, JSONObject formData) throws FormException {
+            ReleaseStep step = (ReleaseStep) super.newInstance(req, formData);
+            // Cf. ParametersDefinitionProperty._doBuild:
+            Object parameter = formData.get("parameter");
+            JSONArray params = parameter != null ? JSONArray.fromObject(parameter) : null;
+            if (params != null) {
+                Jenkins jenkins = Jenkins.getInstance();
+                Job<?,?> context = StaplerReferer.findItemFromRequest(Job.class);
+                Job<?,?> job = jenkins != null ? jenkins.getItem(step.getJob(), context, Job.class) : null;
+
+                AbstractProject project = Jenkins.getActiveInstance()
```

Comment:
🐛 for AbstractProject. See the comment above. Here it's better to just retrieve BuilldableItem or just Item